src/main/peloton/peloton.cpp

```diff
@@ -32,11 +32,16 @@ int main(int argc, char *argv[]) {
     ::google::HandleCommandLineHelpFlags();
   }
 
-  // Print settings
-  if (peloton::settings::SettingsManager::GetBool(
-          peloton::settings::SettingId::display_settings)) {
-    auto &settings = peloton::settings::SettingsManager::GetInstance();
-    settings.ShowInfo();
+  try {
+    // Print settings
+    if (peloton::settings::SettingsManager::GetBool(
+            peloton::settings::SettingId::display_settings)) {
+      auto &settings = peloton::settings::SettingsManager::GetInstance();
+      settings.ShowInfo();
+    }
+  } catch (peloton::SettingsException &exception) {
+    std::cerr << "Cannot load settings. Failed with " << exception.GetMessage() << std::endl;
```

Comment:
I believe we don't use stdout or stderr. This should be a log statement, probably a LOG_ERROR, if it is being treated as a fatal error.

Reply:
Hmm, I wanted to write this error to 'real' stderr, not stdout which is what LOG_ERROR is currently doing. I can change it to use LOG_ERROR for consistency sake but ...

Reply:
You don't want to write to stdout or stderr. Consider Postgres or MySQL or any "service" that is running. Output to stdout or stderr is either not generated or sent to /dev/null. All the information that these processes wish a user to see, is done via logging.

Reply:
Okay, I was thinking from the point of view of peloton running as a systemd service or something like that. Logging errors to stderr would then be easier for filtering. Similar to how apache logs goes to error_logs and other things go to access_logs. But you are probably correct. I am updating the PR with a new commit.